modules/component/src/componentNext.ts

```diff
@@ -138,6 +151,24 @@ export class ComponentNext<P1 extends ComponentProps> {
     )
   }
 
+  /**
+   * Adds ability to return side effect cauasing action matching action type
```

Comment:
"side effect causing action" is a misleading phrase because it could be read as `side effect-causing action` whereas you intend it as `side-effect causing action`

moreover, the emitted action need not cause a side-effect. It's just an OAction... what it does is defined elsewhere.